lib/ash/subject.ex

```diff
@@ -174,32 +166,24 @@ defmodule Ash.Subject do
   @spec get_argument(subject :: t(), argument :: atom() | binary(), default :: term()) :: term()
   def get_argument(subject, argument, default \\ nil)
 
-  def get_argument(subject, argument, default) when is_atom(argument) do
-    value =
-      if Map.has_key?(subject.arguments, argument) do
-        Map.get(subject.arguments, argument)
-      else
-        Map.get(subject.arguments, to_string(argument))
-      end
-
-    if is_nil(value) do
-      default
-    else
-      value
+  def get_argument(%Ash.Changeset{} = subject, argument, default) do
+    case Ash.Changeset.get_argument(subject, argument) do
+      nil -> default
```

Comment:
Actually this isn't right. `get_argument` with a default is different.

```elixir
case fetch_argument(subject, arg) do
  {:ok, v} - v
   :error -> nil
end
```

Reply:
I don't understand what you mean.  Are you saying you don't want the default?  The subject type modules `get_argument` don't currently support providing a default, so i just put it in `Ash.Subject`.  Or are you saying you'd prefer `get_argument` to call `mod.fetch_argument`, match on the result, and return the result when there is one and the default when nil?

Reply:
Sorry, yeah my bad, I messed up the ecample. it should return the default on `:error`, but you figured it out already 😄

Reply:
This is important because if the user provided the arg as `nil`, we shouldn't use the default.

Reply:
Good call.  I'll have the `get_argument/2` delegate to the subject type modules.  The `get_argument/3` will use `fetch_argument/2` for the subject type modules and return the default if an error is returned.